counterpartylib/lib/backend/addrindex.py

```diff
@@ -234,18 +234,19 @@ def getrawtransaction(tx_hash, verbose=False, skip_missing=False):
 def getrawmempool():
     return rpc('getrawmempool', [])
 
-def fee_per_kb(nblocks):
+def fee_per_kb(conf_target, mode):
     """
-    :param nblocks:
+    :param conf_target:
+    :param mode:
     :return: fee_per_kb in satoshis, or None when unable to determine
     """
 
-    # we need to loop because sometimes bitcoind can't estimate a certain nblocks
+    # we need to loop because sometimes bitcoind can't estimate a certain conf_target
     retry = 0
     feeperkb = -1
     while feeperkb == -1:
```

Comment:
estimatesmartfee reports errors, as I understand it, whereas estimatefee would return -1.

Reply:
If someone reviewing this could suggest a solution that would be great. I'm not that familiar with python.

Error found here: https://github.com/bitcoin/bitcoin/pull/10707/files#diff-9651347c8e00bed3ddc7631de569406dR857

Here is one possible way that just trades error messages for error message.

```
def fee_per_kb(conf_target, mode):
    """
    :param conf_target:
    :param mode:
    :return: fee_per_kb in satoshis, or None when unable to determine
    """

    # we need to loop because sometimes bitcoind can't estimate a certain conf_target
    retry = 0
    feeperkb = 'Insufficient data or no feerate found'
    while feeperkb == 'Insufficient data or no feerate found':
        feeperkb = rpc('estimatesmartfee', [conf_target], mode)
        conf_target += 1
        retry += 1

        if retry > 10:
            return None

    return int(feeperkb * config.UNIT)
```

Here is a less optimistic approach that does not make 10 attempts.

```
def fee_per_kb(conf_target, mode):
    """
    :param conf_target:
    :param mode:
    :return: fee_per_kb in satoshis, or None when unable to determine
    """

    feeperkb = rpc('estimatesmartfee', [conf_target], mode)

    if feeperkb == 'Insufficient data or no feerate found':
            return None

    return int(feeperkb * config.UNIT)
```

Reply:
I committed the second option, in the meantime.